docs(js): Document tunnel option and how it helps with adblockers (#3631)
getsentry/sentry-docs: src/platforms/javascript/common/troubleshooting/index.mdx

## Dealing with Ad-Blockers

### Using the `tunnel` Option

> Starting with version `6.5.0` of the JavaScript SDK, you can use the `tunnel` option to tell the SDK to deliver events to the configured URL, instead of using the DSN. This allows the SDK to remove `sentry_key` from the query parameters, which is one of the main reasons ad-blockers prevent sending events. This option also stops the SDK from sending preflight requests, which was one of the requirements that necessitated sending the `sentry_key` in the query parameters.

Author: ```suggestion
Starting with version `6.7.0` of the JavaScript SDK, you can use the `tunnel` option to tell the SDK to deliver events to the configured URL, instead of using the DSN. This allows the SDK to remove `sentry_key` from the query parameters, which is one of the main reasons ad-blockers prevent sending events. This option also stops the SDK from sending preflight requests, which was one of the requirements that necessitated sending the `sentry_key` in the query parameters.
```

To enable it, provide either a relative, or an absolute URL as the `tunnel` option in your `Sentry.init` call:

```js
Sentry.init({
  dsn: "___PUBLIC_DSN___",
  tunnel: "/tunnel",
});
```

Once configured, all events will be sent to the `/tunnel` endpoint. This solution, however, requires an additional configuration on the server, as the events now need to be parsed and redirected to Sentry. Here is an example code written in PHP:

```php
<?php
$envelope = stream_get_contents(STDIN);
$pieces = explode("\n", $envelope, 2);
$header = json_decode($pieces[0], true);
if (isset($header["dsn"])) {
    $dsn = parse_url($header["dsn"]);
    $project_id = intval(trim($dsn["path"], "/"));
    $options = array(
        'http' => array(
            'header'  => "Content-type: application/octet-stream\r\n",
            'method'  => 'POST',
            'content' => $envelope
        )
    );
    echo file_get_contents(
        "https://{$dsn['host']}/api/$project_id/envelope/",
        false,
        stream_context_create($options));
}
```

> With this setup, no events should ever be blocked by ad-blockers, as they are never sent to a third-party URL.

Reviewer: ```suggestion
With this set up, events should not be blocked by ad-blockers, as they are never sent to a third-party URL.
```
